app/controllers/finish_installation_controller.rb

```diff
@@ -70,6 +83,53 @@ def find_allowed_emails
     GlobalSetting.developer_emails.split(",").map(&:strip)
   end
 
+  def setup_discourse_id
+    begin
+      if find_allowed_emails.empty?
+        raise StandardError.new(
+                "No allowed emails configured in DISCOURSE_DEVELOPER_EMAILS. Cannot continue with site setup. ",
+              )
+      end
+      SiteSetting.enable_discourse_id = true
+      # Since we're setting up Discourse ID, disable local logins
+      SiteSetting.enable_local_logins = false
+      # Let ID set people's usernames
+      SiteSetting.auth_overrides_username = true
+      @discourse_id_enabled = true
+      @discourse_id_error = nil
+    rescue StandardError => e
+      @discourse_id_enabled = false
+      @discourse_id_error = e.message
+    end
+  end
+
+  def create_admin_users
+    allowed_emails = find_allowed_emails
+    if allowed_emails.empty?
+      raise StandardError.new(
+              "No allowed emails configured in DISCOURSE_DEVELOPER_EMAILS. Cannot continue with site setup. ",
+            )
+    end
+
+    allowed_emails.each do |email|
+      next if User.find_by_email(email)
+
+      username = UserNameSuggester.suggest(email)
+
+      user =
+        User.new(
+          email: email,
+          username: username,
+          # no password needed, users will login via Discourse ID
+          active: false, # will be activated upon first login
+          admin: true,
+          trust_level: TrustLevel[4],
+        )
+      user.save!(validate: false)
+      Group.refresh_automatic_groups!(:staff, :admins)
```

Comment:
Should we be doing this once _outside_ the loop instead of for every admin users?